server/middlewares/passport.ts

```diff
@@ -18,12 +19,28 @@ export default function createMiddleware(providerName: string) {
 
           if (err.id) {
             const notice = err.id.replace(/_/g, "-");
-            const hasQueryString = err.redirectUrl?.includes("?");
+            const redirectUrl = err.redirectUrl ?? "/";
+            const hasQueryString = redirectUrl?.includes("?");
+
+            // Every authentication action is routed through the apex domain.
+            // But when there is an error, we want to redirect the user on the
+            // same domain or subdomain that they originated from (found in state).
```

Comment:
🥦 move all of the redirection logic into the middleware so that users remain on the same domain when errors occur during auth